com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/flowable/FlowableFacade.java

```diff
@@ -354,4 +368,22 @@ public boolean isJobExecutorActive() {
     public ProcessEngine getProcessEngine() {
         return processEngine;
     }
+
+    @SuppressWarnings("unchecked")
+    public <T> T getVariableFromHistoricVariables(List<HistoricVariableInstance> historicVariablesForProcess, String varName) {
+        return (T) historicVariablesForProcess.stream()
+            .filter(historicVariable -> historicVariable.getVariableName()
+                .equals(varName))
+            .findFirst()
+            .map(HistoricVariableInstance::getValue)
+            .orElse(null);
+    }
+
+    public String findHistoricProcessInstanceIdByProcessDefinitionIdStartsWith(String processInstanceId, String processDefinitionId) {
```

Comment:
Remove this method by filtering the operations by type in ApplicationColorDetector (line 36) before trying to get their variables.